python/voxel_mapper.py

```diff
@@ -5,18 +5,43 @@
 import  numpy as np
 import numbers
 import argparse
-
+from io import BytesIO
+from nibabel import FileHolder, Nifti1Image
+from gzip import GzipFile
+from hdfs import Config
 
         
 def mapper(bin_size, data_max, data_min):
     for line in sys.stdin:
             line = line.strip()
             line = line.split()
             slice_file = os.path.join(line[0])
-            
 
-            #open nifti image
-            slice = nib.load(slice_file)
+            #if image is located in hdfs of not (ie stored locally)
+            hdfs_file = True if line[1] == "True" else False
```

Comment:
ahh ok. I thought that perhaps you might have also been referring to the fact that it was not only bad enough that i was querying the local fs to determine if the file is there in or not, but that I was also doing it twice, especially considering that I should already have established on which fs the file was on. I suppose it doesn't make much of a difference unless between the two mapreduce functions, a file with the same filename and filepath is added and we switch from hdfs to local fs. 

I will add the class and fix this!